sdk/go/common/util/gitutil/git.go

```diff
@@ -326,6 +340,43 @@ func GitCloneOrPull(url string, referenceName plumbing.ReferenceName, path strin
 	return nil
 }
 
+// gitCloneFallback uses the `git` command to clone repositories instead of go-git as
+// go-git does not work with Azure DevOps repos https://github.com/go-git/go-git/issues/64
+func gitCloneFallback(url string, referenceName plumbing.ReferenceName, path string, shallow bool) error {
+	// For shallow clones, use a depth of 1.
+	args := []string{
```

Comment:
Doesn't this also need to handle the pull case?

Reply:
Non-pulumi templates are put into temp directories and marked as ShouldDelete.
https://github.com/pulumi/pulumi/blob/bce637624e00b610c213f4f60fe1dd17aa8513b2/sdk/go/common/workspace/templates.go#L274-L297

Pull is only necessary for persisted directories as they need to be kept up to date. There are no pulumi templates on `dev.azure.com` so the `Pull` portion of the codepath will never run(unless we move our official templates to `dev.azure.com`)

Should I add feature parity for this helper function if the codepath will never run for completeness? I can't funnel it through the other method because go-git has an existing issue with it and I'm not sure if I want to risk modifying behavior of `pulumi new` if it works already.

Depending on your feedback, I'll rename methods and update comments to reflect these assumptions or add the extra `git pull` functionality.

Reply:
this is a draft and I have a new approach described in https://github.com/pulumi/pulumi/issues/6360#issuecomment-1252725559

Reply:
I have added git pull support. The codepath isn't hit in normal running, but I tested it by defaulting pulumi to the system git implementation and wiping the PULUMI_HOME templates directory.